apps/extension/src/pages/formatting.tsx

```diff
@@ -0,0 +1,287 @@
+import { DateFormat, IntroSentenceType } from "@agreeto/api/types";
+import {
+  DEFAULT_LANGUAGE_FORMAT,
+  LANGUAGE_FORMATS,
+  type LanguageFormatItem,
+  extractTextFromSlots,
+} from "@agreeto/calendar-core";
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { addHours, format, startOfDay } from "date-fns";
+import { useEffect, useRef, useState } from "react";
+import { HiCheckCircle } from "react-icons/hi";
+import { TiArrowSortedDown } from "react-icons/ti";
+
+import { trpcApi } from "~features/trpc/api/hooks";
+
+// @ts-expect-error - svg's are not typed
+import previewImage from "./../../assets/gmail-preview.svg";
+
+export const Formatting = () => {
+  const utils = trpcApi.useContext();
+  const introSentenceRef = useRef<HTMLTextAreaElement>(null);
+  const [languageDropdownOpen, setLanguageDropdownOpen] = useState(false);
+  const [dateFormatDropdownOpen, setDateFormatDropdownOpen] = useState(false);
+  const [introSentenceFocused, setIntroSentenceFocused] = useState(false);
+  const [canSave, setCanSave] = useState(false);
+  const [introSentence, setIntroSentence] = useState<string>(
+    DEFAULT_LANGUAGE_FORMAT.defaultIntroSentence,
+  );
+  const [introSentenceType, setIntroSentenceType] = useState<IntroSentenceType>(
+    IntroSentenceType.DEFAULT,
+  );
+  const [selectedLanguage, setSelectedLanguage] = useState<LanguageFormatItem>(
+    DEFAULT_LANGUAGE_FORMAT,
+  );
+  const [selectedDateFormat, setSelectedDateFormat] = useState<DateFormat>(
+    DateFormat.MMMM_d_EEEE,
+  );
+
+  const { data: formattings } = trpcApi.formatting.byCurrentUser.useQuery();
+  const { mutate: updateFormatting, isLoading: isUpdating } =
+    trpcApi.formatting.update.useMutation({
+      onSuccess() {
+        utils.formatting.byCurrentUser.invalidate();
+      },
+    });
+
+  useEffect(() => {
+    const formatting = formattings?.find(
```

Comment:
trpc procedures are cheap, can make a separate procedure like `byLanguage` and just send the language in, and select a single formatting from the db that match the language there, instead of sending all of them back and filtering here

Reply:
I would prefer no delay there for UX. What do you think @p6l-richard? Correct me if I am wrong but as far as I understand, API calls take longer when made from extension. Like when you change the language by dropdown on top of the availability text-area, it takes too long. Idk why.

This is the dropdown place in old version
<img width="310" alt="Screen Shot 2022-12-01 at 10 16 35" src="https://user-images.githubusercontent.com/30385360/204989620-f36f71c4-bd7f-4035-9d8e-4ae5fe5e1c0d.png">

Reply:
So, I had to stop my work in this repo and started to work on the old one to complete the formatting page completely. What is missing is the `Advanced Formatting` feature. After I complete that part, I will be update this PR with CRs

Reply:
I agree with @bgavas. We have to find a way to not delay the UX. 

I still think that this should be handled server-side though as this is our custom "availability appearance" logic and doesn't have anything to do with the react-calendar per se. Therefore it does feel misplaced somewhat (I am responsible for the location of this in the calendar-react package by the way). 

Since we don't have an @agreeto/availability-core and @agreeto/availability-react package yet, I think putting this on our server is acceptable. 

However, we need to find a way not to delay the UX. Therefore, I think we need to prefetch this during the hydration of our app. Whenever the user opens the react app for the first time, we could fetch this data along with other required data on the calendar route (eg events, verify session etc). 

The downside I see with an approach like this is that if we're not careful about structuring our app properly in our routes, we may run into waterfalls of a lot of requests happening. 
Since we're using tan stack router and have trpc batching implemented, I think we're well equipped to avoid the waterfalls.

We just need to pay attention to not stuff too many requests into our hydration over time. 

The good thing: this data needs to be fetched only once per component lifetime and won't change when changing weeks for example. So I do consider it cheaper than the events request and would find the server-side location placement acceptable if we find a way not to delay the UX for loading the options. 

The same applies to the co worker search bar by the way. 

I think we could prefetch user.getFriends ahead of time by simply reading common friends from the events of the current and/or past week. 
Out of scope of this PR though.